### PR TITLE
Julia Runner and Zygote module

### PR DESCRIPTION
**Pull request by msdmkats**
Julia runner and zygote module supporting GMM problem.

### CODE REVIEW

File: JuliaManifest.toml [hunk 1]
```diff
@@ -0,0 +1,273 @@
+# This file is machine-generated - editing it directly is not advised
```
[the remark this answers is not in the document]
Comment (msdmkats) [reply JuliaManifest.toml]: Yes. It makes sure everyone uses the same versions of all julia packages.

File: src/julia/modules/zygote/ZygoteGMM.jl [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+export get_gmm_test
+
+"Computes logsumexp. Input should be 1 dimensional"
+function logsumexp(x)
+    mx = maximum(x)
+    log(sum(exp.(x .- mx))) + mx
+end
+
+sumsq(v) = sum(abs2, v)
```
[the remark this answers is not in the document]
Comment (msdmkats) [reply src/julia/modules/zygote/ZygoteGMM.jl]: Removed before reading this comment - it wasn't used at all.

File: ADBench/run-all.ps1 [hunk 3]
```diff
@@ -440,11 +449,12 @@ $tool_descriptors = @(
     [Tool]::new("Manual", "cpp", [ObjectiveType] "GMM, BA, Hand, LSTM", $false, 0.0)
     [Tool]::new("ManualEigen", "cpp", [ObjectiveType] "GMM, BA, Hand, LSTM", $true, $default_tolerance)
     [Tool]::new("ManualEigenVector", "cpp", [ObjectiveType] "GMM", $true, $default_tolerance)
-    #[Tool]::new("Tapenade", "cpp", [ObjectiveType] "BA, LSTM, GMM, Hand", $true, $default_tolerance)
-    #[Tool]::new("DiffSharp", "bin", [ObjectiveType] "BA", $false, 0.0, $true, $false)
-    #[Tool]::new("Autograd", "py", [ObjectiveType] "GMM, BA", $false, 0.0, $true, $false)
-    #[Tool]::new("PyTorch", "py", [ObjectiveType] "GMM, LSTM", $false, 0.0)
-    #[Tool]::new("Julia", "julia", [ObjectiveType] "GMM, BA", $false, 0.0)
+    [Tool]::new("Tapenade", "cpp", [ObjectiveType] "BA, LSTM, GMM, Hand", $true, $default_tolerance)
```
[the remark this answers is not in the document]
Comment (msdmkats) [reply ADBench/run-all.ps1]: They used to be uncommented 😉 The reason they are commented in master is that someone forgot to uncomment them after debugging.

File: src/julia/modules/zygote/ZygoteGMM.jl [hunk 4]
```diff
@@ -0,0 +1,219 @@
+module ZygoteGMM
+include("../../shared/load.jl")
+using ADPerfTest
+using GMMData
+using Zygote
+using Zygote: @adjoint
```
[the remark this answers is not in the document]
Comment (msdmkats) [reply src/julia/modules/zygote/ZygoteGMM.jl]: Sorry, I don't really understand what this refers to. Was this comment added before some force-push?

[the remark this answers is not in the document]
Comment (awf) [reply src/julia/modules/zygote/ZygoteGMM.jl]: Just that we're importing adjoint in order to allow @adjoint macro, but I would consider importing gradient too, so the Zygote.gradient line later is replaced by gradient.  It's a matter of taste, and normally I would go the other way, but here I believe importing one suggests we should import the other.

[the remark this answers is not in the document]
Comment (msdmkats) [reply src/julia/modules/zygote/ZygoteGMM.jl]: Ah. Ok. There were a bunch of other qualified uses of functions from Zygote module, so for uniformity I made `@adjoint` uses qualified as well.

**Comment by msdmkats** [comment]
Finally, plots:
GMM (1k)  Jacobian  - Release Graph
GMM (10k)  Jacobian  - Release Graph

On my machine Zygote produced a lot of NaNs instead of answers for `d = 64` and some cases of `d = 32`. These are shown as red triangles (I used plotting scripts from toelli/graph_correctness_main branch). This may be related to the fact that my machine has only 16 GB of RAM, because on large problems it demonstrated symptoms of being out of it.

Would be nice if someone ran these on a machine with 32 GB of RAM.

**Comment by awf** [comment]
Great, thanks!

**Pull request merged**